### ISSUE TITLE
dhclient overwrites /etc/resolv.conf with last response received

### ISSUE THREAD

**Issue by mpictor** [u-root/u-root#1837]
Regarding `pkg/dhclient` -
The way it's used in cmds/boot/pxeboot and cmds/core/dhclient is like
```go
    r := dhclient.SendRequests( /*...*/ )
    for result := range r {
        //...
        result.Lease.Configure()
        //...
    }
```
`result.Lease.Configure()` calls `dhclient.WriteDNSSettings()`. That function overwrites `/etc/resolv.conf`, ignoring existing values - so the order in which responses are processed (and presumably the order in which they're received) determines the content of `/etc/resolv.conf`.

@hugelgupf says:
> It was acceptable at the time, but I think you're right, we shouldn't be doing that.

**Comment by 0utsights** [comment]
The fix is ready in the [proposed pull request](https://github.com/u-root/u-root/pulls), which preserves resolver entries across DHCP lease updates and includes regression coverage.
https://github.com/u-root/u-root/pull/3682

**Comment by rminnich** [comment]
I don't think this is an issue. If you reconnect your DHCP-configured device to another network, there's a good chance you want resolve.conf changed to reflect that change. I am closing this issue until someone actually hits a problem with it. No one has ever reported this problem.
